src/internal/scheduler/animationFrameProvider.ts

```diff
@@ -1,7 +1,7 @@
 /** @prettier */
 import { Subscription } from '../Subscription';
 
-type AnimationFrameProvider = {
+interface AnimationFrameProvider {
```

Comment:
thought: If this change from a `type` to an `interface` was effected by a lint rule, I would switch it off. Increasingly, types are able to due more and more. Personally, I would favour using interfaces only when they offer a feature that's not offered by a type. If it's a rule, I think it's a useless rule. Mostly because TypeScript has changed so much in the last few years.

Reply:
Hmmm... for me this was about just homogenizing the approach. FWIW: I think if we're just defining an interface (like one for a provider) than it should probably be an interface. However, if we're defining some sort of argument type that could be more than one thing, or a function shape, then I guess I'd prefer a `type`. But I suppose it's all personal preference.